File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/ConstantExpression.cs
using Assignement7.Collection;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Model
{
    public class ConstantExpression : Expression
    {
        private int value;

        public ConstantExpression(int val)
        {
            this.value = val;
        }

        public override int Evaluate(MyIDictionary<string, int> symbolTable)
        {
            return value;
        }

        public override string ToString()
        {
            return value.ToString();
        }
    }
}

File: Semester3/Advanced Methods of Programming/ExamenMAP/src/model/expressions/ConstantExpression.java
package model.expressions;

import myCollections.MyIDictionary;

public class ConstantExpression extends Expression {
    private Integer value;

    public ConstantExpression(Integer value)
    {
        this.value = value;
    }

    @Override
    public Integer evaluate(MyIDictionary<String, Integer> symbolTable, MyIDictionary<Integer, Integer> heapTable) {
        return value;
    }

    @Override
    public String toString() {
        return value.toString();
    }
}
